staging/src/k8s.io/apiserver/pkg/apis/audit/validation/validation.go

```diff
@@ -60,3 +65,40 @@ func validateLevel(level audit.Level, fldPath *field.Path) field.ErrorList {
 		return field.ErrorList{field.NotSupported(fldPath, level, validLevels)}
 	}
 }
+
+func validateNonResourceURLs(urls []string, fldPath *field.Path) field.ErrorList {
+	var allErrs field.ErrorList
+	for i, url := range urls {
+		if url == "*" {
+			continue
+		}
+
+		if !strings.HasPrefix(url, "/") {
+			allErrs = append(allErrs, field.Invalid(fldPath.Index(i), url, "invalid non-resource URL format"))
+		}
+
+		if strings.ContainsRune(url[:len(url)-1], '*') {
+			allErrs = append(allErrs, field.Invalid(fldPath.Index(i), url, "invalid non-resource URL format, only support one trailing * at the end"))
```

Comment:
nit: to keep this error consistent with the wording of other validation rules maybe "non-resource URL wildcards '*' must be the final character of the rule"